### DISCUSSION TITLE
Terminal based admin prompt?

### DISCUSSION THREAD

**Discussion by svin24** [microsoft/winget-cli#5077] [General]
### Description of the new feature / enhancement

This is probably not the place to ask for this as this may affect way more windows components than just winget.

I was just thinking that many times when installing an application a hidden admin privileges prompt pops up in the background and i think winget is frozen, completely ignoring the flashing icon on the bottom.

Instead of that i think a terminal based request for admin permissions dialog would be quite cool

### Proposed technical implementation details

I took a text sample and edited it in neovim to show how it would look like

Kind of like the ones found on linux, except those ones are for confirming the installation and not administration privileges elevation 

![Image](https://github.com/user-attachments/assets/fa482df2-b726-41f6-9cba-f8017a1cf1ae)

**Comment by florelis** [comment]
It is not a prompt, but if we know that a package requires admin to install, we show a warning instructing to expect the UAC prompt. If the warning isn't showing for a package, it should be added to the package metadata in winget-pkgs.

I wonder if we could do something interesting using [Sudo for Windows](https://github.com/microsoft/sudo) to elevate within the same terminal window (assuming the user's  settings allow it). Or maybe we could add a winget setting to skip any packages that require admin instead of waiting for the user to accept the prompt 🤔

**Comment by denelon** [comment]
The main challenge here is that the installers themselves in some cases are requesting the elevation, and focus may be lost from the UAC (User Account Control) which is why it ends up down in the taskbar waiting for interaction.

When WinGet is aware of an impending prompt it will inform the user, and as @florelis mentioned the manifest can be updated to accommodate that scenario. When WinGet is acting as an installer or an orchestrator, it can trigger the prompt which in most cases will ensure focus for the UAC prompt isn't lost.

**Comment by denelon** [comment]
Many users will default to running install or upgrade elevated which also doesn't always work. Some packages prohibit elevated install, and many will select the scope "user" vs. "system/machine" based on how they are executed. In some cases, installers will also perform their own checks on the user/account performing the installation and based on whether that user is a member of the local administrator group, that will determine the scope.

**Comment by denelon** [comment]
Rather than just closing this "Issue" as "won't do/can't do" I'll go ahead and convert it to a discussion. We have several active community members who are improving manifests. As you run into packages where this is happening, you could file an issue over at winget-pkgs to request the manifest is updated, or you could raise the PR yourself.
